code/script.js

```diff
@@ -225,101 +230,175 @@ const generateBoard = () => {
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  console.log(secret)
+  }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
+  winOrLose.style.display = 'none'
+  board.style.display = 'flex' 
   charactersInPlay = CHARACTERS
+  setSecret()
+  generateBoard()
   // What else should happen when we start the game?
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label; 
+  const value = questions.options[questions.selectedIndex].value;  
 
+  
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
   // const value =
 
-  currentQuestion = {
-    category: category,
-    // value: value
+
+  if (category === 'vices') {
+    currentQuestion = {
+      attribute: 'vices', 
+      value: value, 
+      category: category, 
+      }
+  } else if (category === 'musictaste') {
+      currentQuestion = {
+        attribute: 'musictaste', 
+        value: value, 
+        category: category,
+      }
+  } else if (category === 'traits'){
+      currentQuestion = {
+        attribute: 'traits', 
+        value: value, 
+        category: category,
+  } } else {
+    currentQuestion = {
+      attribute: value, 
+      value: true, 
+      category: category,
+    }
+   }
   }
-}
+
 
 // This function should be invoked when you click on 'Find Out' button.
```

Comment:
Clear comment! I would say that all your comments are straightforward and describe the functions well.